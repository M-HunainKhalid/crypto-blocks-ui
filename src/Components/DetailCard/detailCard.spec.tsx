import { render, screen } from "@testing-library/react";
import { DetailCard } from ".";

describe("Detail Card", () => {
  const props = {
    detailName: "Sample Detail",
    detailValue: "Sample Value",
  };
  it("should render detail card component correctly", () => {
    const detailCard = render(<DetailCard {...props} />);
    expect(screen.getByText(props.detailName)).toBeInTheDocument();
    expect(screen.getByText(props.detailValue)).toBeInTheDocument();
    expect(detailCard).toMatchSnapshot();
  });
});
