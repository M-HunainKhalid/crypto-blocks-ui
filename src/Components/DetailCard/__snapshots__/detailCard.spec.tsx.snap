// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Card should render detail card component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation3 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <h6
            class="MuiTypography-root makeStyles-detailName-2 MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Sample Detail
          </h6>
          <p
            class="MuiTypography-root makeStyles-detailValue-3 MuiTypography-body1"
          >
            Sample Value
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <h6
          class="MuiTypography-root makeStyles-detailName-2 MuiTypography-h6 MuiTypography-gutterBottom"
        >
          Sample Detail
        </h6>
        <p
          class="MuiTypography-root makeStyles-detailValue-3 MuiTypography-body1"
        >
          Sample Value
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
